internal/command/docker.go

```diff
@@ -137,9 +163,21 @@ func goBuild(ctx Context) error {
 
 	// add ldflags to command, if any
 	if len(ldflags) > 0 {
-		args = append(args, "-ldflags", fmt.Sprintf("'%s'", strings.Join(ldflags, " ")))
-	}
+		flags := make([]string, len(ldflags))
+		for i, flag := range ldflags {
+			flags[i] = fmt.Sprintf("-ldflags=%s", flag)
+		}
 
+		// ensure that GOFLAGS is not overwritten as they can be passed
+		// by he --env argument
+		if idx := findGoFlags(ctx); idx > -1 {
+			// append the ldflags after the existing GOFLAGS
+			ctx.Env[idx] += " " + strings.Join(flags, " ")
```

Comment:
Seems like a combination of two different ways. I would suggest either doing `" " + flags + " "` or perhaps even `strings.Join(" ", flags, " ")`.

Reply:
Ah, I thought that `strings.Join()` did something different. Ignore this one.